packages/api/src/trpc.ts

```diff
@@ -107,13 +115,27 @@ export const publicProcedure = t.procedure;
  *
  */
 
-const isAuthed = t.middleware(({ next, ctx }) => {
+const isAuthed = t.middleware(async ({ next, ctx }) => {
   if (!ctx.auth.userId) {
     throw new TRPCError({ code: "UNAUTHORIZED" });
   }
+
+  // If the user isn't in our database yet, insert them
+  const userId = ctx.auth.userId;
+  const user = await ctx.db.user.upsert({
```

Comment:
Since we're already going to fetch the userId and append it to the context, we might as well do the insertion here as well if they don't exist